### DISCUSSION TITLE
Multi-step modal footer UX

### DISCUSSION THREAD

**Discussion by vankhoawin** [twilio-labs/paste#1262] [Q&A] [answered]
Hi team, there's a growing amount of multi-step modals and I've seen a few ways of implementation, so I'd love to start a discussion and nip this issue in the bud before we have too many divergent patterns.

- Back and Cancel buttons in the Modal Footer: I've seen two main patterns of Modal footers for multi-step modals so far (see below screenshots).
- Adding a confirmation popover (or something along those lines) to confirm user's choice to exit a multi-step modal: related to the above bullet, I've noticed some designs have a Cancel button where I'd personally expect a Back button. Without confirmation it is easy for a user to exit out of a multi-step modal without the possibility of resuming that flow.
- Anything else? cc @andipants12 @ishah-twilio

**Variation 1**: Cancel button always visible, Back button included after first step

![image](https://user-images.githubusercontent.com/6936736/110375290-a321ea00-8006-11eb-90b8-e0b5a7220fc8.png)

**Variation 2**: Cancel button visible only on first step (first screenshot), Back button visible on the other steps (second screenshot)

![image](https://user-images.githubusercontent.com/6936736/110375461-d9f80000-8006-11eb-8e0f-68ea3f656418.png)
![image](https://user-images.githubusercontent.com/6936736/110375488-e11f0e00-8006-11eb-94b2-50f60116c807.png)

**Comment by ishah-twilio** [comment]
@vnguyen94 - IoT team had followed following paste component

https://paste.twilio.design/components/modal/#footer-actions-directional-alignment

It tasted very well in the usability studies for IoT Customers. Let me know if you need any specific information or how may I help.

**Comment by corinne-nagel** [comment]
Hey @vnguyen94 , thanks for calling this out. We'll work on getting alignment on this and bring it to the committee. I'll share what comes from that meeting here. Thanks again!

**Reply by vankhoawin** [reply]
Thanks! and let me know if I should be bringing these up in office hours or some other way, I saw value in it being written down.

**Comment by loreina** [comment, marked as answer]
Closing this discussion due to inactivity. The general recommendation would be to keep the "Back" button as a left action, and "Cancel" as a secondary button next to the primary action.

<img width="740" alt="image" src="https://user-images.githubusercontent.com/23108291/199101305-ed90dfee-d4ed-42d1-bd03-527ebc2af57e.png">
